### PR TITLE
Build ROS Kinetic from awesomebytes/roogp

### PR DESCRIPTION
**Pull request by awesomebytes**
As per https://github.com/softbankrobotics-research/sbre_robot_ros_gentoo_prefix/issues/4

You may want to squash the commits together on merging.

Note that the image starts from my 'desktop' image as it gives the advantage of having some slow-compiling elements like PyQt already built. It minimizes re-implementation.

It also features naoqi_libqi, naoqi_libqicore and naoqi_driver building with workarounds. These should instead be PR'ed into ros-overlay.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    variables:
+      docker_image_name: sbre_os_kinetic
+
+    steps:
+    - script: cd ros_base; docker build --network host -f Dockerfile.from_roogp -t $(DockerHub_user)/$(docker_image_name):$(Build.BuildId) .
+      displayName: 'docker build'
+      # This means up to 6h
+      timeoutInMinutes: 0
+    - script: docker login -u $(DockerHub_user) -p $(DockerHub_password)
```
Comment (awesomebytes) [inline azure-pipelines.yml:17]: Add DockerHub_user, DockerHub_password (as secret) in the Azure Pipelines "Variables" section for the pipeline.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        export RELEASE_FILE_NAME=sbre_os_$(rosdistro)-${PLATFORM}_${DATE}.tar.lzma
+        export RELEASE_NAME="SBRE OS based on Gentoo Prefix and ROS Kinetic"
+        export BODY="Extract with: cat ${RELEASE_FILE_NAME}.part-* > ${RELEASE_FILE_NAME}; tar xvf sbre_os_*.tar.lzma; ./gentoo/startprefix"
+        mv $HOME/sbre_os.tar.lzma ${RELEASE_FILE_NAME}
+        docker system prune -a -f
+        split -b 1024MB -d ${RELEASE_FILE_NAME} ${RELEASE_FILE_NAME}.part-
+        rm ${RELEASE_FILE_NAME}
+        export PATH=$HOME/.local/bin:$PATH
+        export PYTHONPATH=$HOME/.local/lib/python2.7/site-packages:$PYTHONPATH
+        githubrelease --github-token $(GitHub_Token) release $(Build.Repository.Name) create "${RELEASE_TAG}" --publish --name "${RELEASE_NAME}" --body "${BODY}" "${RELEASE_FILE_NAME}.part-*"
```
Comment (awesomebytes) [inline azure-pipelines.yml:100]: Add GitHub_Token (as secret) in the Azure Pipelines "Variables" section for the pipeline.

File: ros_base/Dockerfile.package_from_roogp [hunk 3]
```diff
@@ -0,0 +1,6 @@
+FROM awesomebytes/sbre_os_kinetic
```
Comment (awesomebytes) [inline ros_base/Dockerfile.package_from_roogp:1]: Change 'awesomebytes' to your DockerHub user.

**Comment by mcaniot** [comment]
I close this PR for the reason mentionned in this comment https://github.com/softbankrobotics-research/sbre_robot_ros_gentoo_prefix/issues/4#issuecomment-605963524

**Pull request closed without merging**